src/modules/select/index.ts

```diff
@@ -65,33 +73,21 @@ function select(
 
         return Promise.resolve(wallet)
       })
-    )
+      // Filter out the skipped injected wallet if it is undefined
+    ).then(wallets => {
+      return wallets.filter(wallet => !!wallet)
+    })
   }
 
-  const initWalletForNetwork = (walletModule: any) =>
-    walletModule.default({ networkId })
-
   return Promise.all(
-    defaultWalletNames.map(walletName =>
-      getModule(walletName).then(initWalletForNetwork)
-    )
-  ).then(modules => {
-    const installedModuleNames = modules.map(m => m.name)
-
-    if (window.ethereum) {
-      const injectedProviderName = getProviderName(window.ethereum)
-
-      if (!installedModuleNames.some(name => name === injectedProviderName)) {
-        // if an injected provider is present, list it as an option
-        // only if it's not already one of the provided options
-        return getModule('injected')
-          .then(initWalletForNetwork)
-          .then(m => [m, ...modules])
-      }
-    }
-
-    return modules
-  })
+    defaultWalletNames
+      .map(walletName =>
+        walletName === 'detectedwallet' && !injectedWalletDetected()
+          ? null
+          : getModule(walletName).then((m: any) => m.default({ networkId }))
+      )
+      .filter(wallet => !!wallet)
+  )
```

Comment:
@taylorjdawson thanks for the updates. in this block and the one above you can do the filtering first so that we don't need the awkward null filtering:

```
return Promise.all(
    defaultWalletNames
      // only include a detected wallet if it's not already one of the provided options
      .filter(walletName => walletName !== 'detectedwallet' || injectedWalletDetected())
      .map(walletName => getModule(walletName).then((m: any) => m.default({ networkId }))
  )
```

Reply:
Good suggestion 👍🏾